### PR TITLE
Fix ROOT->GCC dependency and allow building AliRoot+O2 together

### PR DESCRIPTION
**Pull request by sawenzel**
* make the dependency of ROOT on GCC more explicit; fixes an issue
   where ROOT was not compiled with the required version of GCC
   (see #741) when compiling with o2 defaults
 * allow for the possibility to build AliRoot with the o2 defaults
 * disable the old explicit dependency of o2 on AliRoot which is no
   longer needed

### CODE REVIEW

**Comment by sawenzel** [comment]
this PR is based on the dev branch. We could cherry-pick this commit into master directly.

**Comment by ktf** [comment]
Given it only touched O2, I rebased this on top of master and fixed the conflict with @matthiasrichter recent commit in `defaults-o2.sh`. What about the other defaults? Can you check if the need the fix as well?

File: defaults-o2.sh [hunk 1]
```diff
@@ -6,7 +6,6 @@ env:
   CMAKE_BUILD_TYPE: "RELWITHDEBINFO"
 disable:
   - AliEn-Runtime
-  - AliRoot
```
Comment (matthiasrichter) [inline defaults-o2.sh]: also `AliEn-Runtime` should be removed for consistency

**Comment by sawenzel** [comment]
Updated this PR, added the GCC dependency for ROOT to more defaults. Also implemented suggestion by @matthiasrichter.

**Comment by sawenzel** [comment]
Please squash the commits (I edited online).

**Comment by sawenzel** [comment]
Closing this. Another PR was made with same content.

**Pull request closed without merging**